proposals/model-management.md

```diff
@@ -0,0 +1,149 @@
+# Model Management
+
+## Problem
+
+This proposal is attempting to solve the following problems:
+
+* Slow model loading on cold-start
+* Storage of LoRA adapters generated from finetuning jobs
+* Dynamic LoRA adapter injection
+
+While working within the following constraints:
+
+* KubeAI should be able to be installed with a single command
+* KubeAI should not rely on external dependencies to provide basic functionality
+* KubeAI should be able to run on a local machine
+* KubeAI should be able to be run on clouds with heterogenous disk features (limited support for ReadOnlyMany)
+
+## Solution
+
+KubeAI should support configurable storage for both models and adapters.
+
+These settings should be configurable on a per Model basis because the optimum caching technique is relative to the size and frequency-of-use of the model. For example: GCP Hyperdisk ML is relatively expensive compared to other options, however it can be highly performant especially when it comes to loading large models. They should also be configurable on a per-finetune-job basis so that users can use KubeAI to publish adapters to their huggingface repos.
+
+### Example Models
+
+```yaml
+# Model configured with adapters from huggingface
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+  adapters:
+  # Adapter can be requested by name at inference time
+  - name: sarcasm
+    url: "hf://myrepo/sarcastic-lora-adapter-for-llama-3-8b-Instruct"
+
+# Model that explicitly asks for a cache profile
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-405B-Instruct"
+  cacheProfile: HyperDisk
+
+# A Model without any cache specified will use the default set in KubeAI's config.yaml
```

Comment:
and the default cacheProfile on installation is cacheProfile: None?

Reply:
yep!

Reply:
When you say "will use the default set in KubeAI's config.yaml" does that mean the model catalog at `charts/kubeai/charts/models/values.yaml`? If so, and a model is used that isn't in the model catalog, will cacheProfile be a required field?